### PR TITLE
Include testing considerations in issue and PR templates

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Include testing considerations in issue and PR templates

#### Changes

- Add expected test scenarios/regression description in issues.
- Add `Testing` section to PR templates and update checklist.

#### Notes for Reviewers

Let's please discuss the wording and see if we can reach a consensus on this. 

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [x] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -50,6 +50,19 @@ Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable
...
+...
+
+##### What else needs to be checked as regression?
+<!--
+Please think about things that this change could affect and how that can be tested?
+You can also @mention experts if you need help with defining this.
+-->
+
+...
+
```
Comment (htdvisser) [inline .github/ISSUE_TEMPLATE/bug_report.md:65]: I don't think bug reporters can answer this at the time they report the bug.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -44,6 +44,22 @@ Paste the output of "ttn-lw-cli version" or "ttn-lw-stack version" if applicable
...
+...
+
+##### What else needs to be checked as regression?
+<!--
+Please think about things that this change could affect and how that can be tested?
+You can also @mention experts if you need help with defining this.
+-->
+
+...
+
```
Comment (htdvisser) [inline .github/ISSUE_TEMPLATE/feature_request.md:62]: I don't think feature requesters can answer this at the time they submit the issue.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -17,6 +17,14 @@ Closes #0000, References #0000, etc.
 - ...
 - ...
 
+
+#### Testing
+
+<!--
+How did you verify that this change works and doesn't cause any regression?
+Check the issue template for expected test scenarios and make sure to add logs/screenshots here that verify those.
```
Comment (htdvisser) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: I think the first line is sufficient. I don't think we should refer to the issue template or ask developers to copy-paste log output. I do like the idea of adding screenshots for frontend changes, but IMO those don't belong in the "testing" section.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -34,7 +42,7 @@ implementation proposal in the referenced issues.
 
 - [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
 - [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md`. The target branch is set to `master` if the changes are fully compatible with existing API, database, configuration and CLI.
-- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well.
+- [ ] Testing: The changes are covered with unit tests. The changes are tested manually as well. The output of tests (screenshots/logs) are added to the `#### Testing` section above.
```
Comment (htdvisser) [inline .github/PULL_REQUEST_TEMPLATE.md:45]: I think the "testing" section replaces this bullet.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 2 above)
Comment (johanstokking) [reply .github/ISSUE_TEMPLATE/feature_request.md:62]: Agreed. In fact, I would move this section to the PR template, as the PR author may have a sense for potential regression elsewhere

**Comment by KrishnaIyer** [comment]
Ok so I've reworded this.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -17,6 +17,19 @@ Closes #0000, References #0000, etc.
 - ...
 - ...
 
+
+#### Testing
+
+<!--
+How did you verify that this change works
+-->
```
Comment (htdvisser) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: Put this on one line, otherwise people might add text inside the comment. Also below.

**Pull request merged**